src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp

```diff
@@ -0,0 +1,281 @@
+#include <OpenMS/FORMAT/FileTypes.h>
+#include <OpenMS/FORMAT/HANDLERS/MzMLHandler.h>
+#include <ostream>
+#include <OpenMS/KERNEL/MSExperiment.h>
+
+#include <OpenMS/ANALYSIS/ID/SiriusMSConverter.h>
+
+#include <QDir>
+
+using namespace OpenMS;
+using namespace std;
+
+namespace OpenMS
+{
+  //Precursor correction (highest intensity)
+  Int getHighestIntensityPeakInMZRange(double test_mz, const MSSpectrum<Peak1D>& spectrum1, double left_tolerance, double right_tolerance)
+  {
+    MSSpectrum<Peak1D>::ConstIterator left = spectrum1.MZBegin(test_mz - left_tolerance);
+    MSSpectrum<Peak1D>::ConstIterator right = spectrum1.MZEnd(test_mz + right_tolerance);
+
+    // no MS1 precursor peak in +- tolerance window found
+    if (left == right || left->getMZ() > test_mz + right_tolerance)
+    {
+      return -1;
+    }
+
+    MSSpectrum<Peak1D>::ConstIterator max_intensity_it = std::max_element(left, right, Peak1D::IntensityLess());
+
+    if (max_intensity_it == right)
+    {
+      return -1;
+    }
+
+    return max_intensity_it - spectrum1.begin();
+  }
+
+
+  String SiriusMSFile::store(const PeakMap &spectra)
+  {
+    OpenMS::String msfile;
+    int count = 0;
+    //check for all spectra at the beginning if spectra are centroided
+    //determine type of spectral data (profile or centroided) - only checking first spectrum (could be ms2 spectrum)
+    SpectrumSettings::SpectrumType spectrum_type = spectra[0].getType();
+
+    if (spectrum_type == SpectrumSettings::RAWDATA)
+    {
+      throw OpenMS::Exception::IllegalArgument(__FILE__, __LINE__, __FUNCTION__, "Error: Profile data provided but centroided spectra are needed. Please use PeakPicker to convert the spectra.");
+    }
+
+    // loop over all spectra in file
+    ofstream os;
+
+    for (PeakMap::ConstIterator s_it = spectra.begin(); s_it != spectra.end(); ++s_it)
+    {
+      // process only MS2 spectra
+      if (s_it->getMSLevel() != 2)
+      {
+      continue;
+      }
+
+      const MSSpectrum<Peak1D>& spectrum = *s_it;
+
+      int scan_index = s_it - spectra.begin();
+
+      const vector<Precursor>& precursor = spectrum.getPrecursors();
+      double collision = precursor[0].getActivationEnergy(); //not working?
+
+      IonSource::Polarity p = spectrum.getInstrumentSettings().getPolarity(); //charge
+
+      // needed later for writing in ms file
+      int int_charge(1);
+
+      if (p == IonSource::Polarity::POSITIVE)
+      {
+        int_charge = +1;
+      }
+      else
+      {
+        LOG_WARN << "SiriusMSConverter (due to Sirius) only support positive ion mode and mono charged analytes." << endl;
+        continue;
+      }
+
+      //there should be only one precursor and MS2 should contain peaks to be considered
+      if (precursor.size() == 1 && spectrum.empty() == false)
+      {
+        //read charge annotated to MS2
+        int precursor_charge = precursor[0].getCharge();
+
+        //sirius only supports +1 charge so far
```

Comment:
As far as I know Sirius supports charges of +1 & -1 thanks for pointing that out.